src/common/store/storage/reducer.ts

```diff
@@ -0,0 +1,48 @@
+import type { Reducer } from '@reduxjs/toolkit';
+import type { AnyAction } from 'redux';
+
+import { rehydrateState } from './persistence.slice';
+
+export interface PersistConfig {
+  key: string;
+  whitelistedActions: string[];
+  whitelistedKeys: string[];
+}
+
+export interface PersistState {
+  _persistence?: {
+    whitelistedActions: string[];
+    whitelistedKeys: string[];
+  };
+}
+
+export const injectPersistent = <A>(
+  { key, ...config }: PersistConfig,
```

Comment:
If we filter out the key here do we actually need it elsewhere or can it be removed?